internal/testtypes/builders.go

```diff
@@ -0,0 +1,525 @@
+	obj.SetAPIVersion(BootstrapGroupVersion.String())
+	obj.SetKind(GenericBootstrapConfigTemplateKind)
+	obj.SetNamespace(b.namespace)
+	obj.SetName(b.name)
+
+	return obj
+}
+
+// InfrastructureClusterTemplateBuilder holds the variables needed to build a template for cluster.spec.infrastructureRef.
+type InfrastructureClusterTemplateBuilder struct {
+	namespace string
+	name      string
+}
+
+// NewInfrastructureClusterTemplateBuilder returns an InfrastructureClusterTemplateBuilder with the given name and namespace.
+func NewInfrastructureClusterTemplateBuilder(namespace, name string) *InfrastructureClusterTemplateBuilder {
+	return &InfrastructureClusterTemplateBuilder{
+		namespace: namespace,
+		name:      name,
+	}
+}
+
+// Build creates a new Unstructured object with the variables passed to the InfrastructreClusterTemplateBuilder.
+func (i *InfrastructureClusterTemplateBuilder) Build() *unstructured.Unstructured {
+	obj := &unstructured.Unstructured{}
+	obj.SetAPIVersion(InfrastructureGroupVersion.String())
+	obj.SetKind(GenericInfrastructureClusterTemplateKind)
+	obj.SetNamespace(i.namespace)
+	obj.SetName(i.name)
+
+	if err := unstructured.SetNestedField(obj.UnstructuredContent(), true, "spec", "template", "spec", "fakeSetting"); err != nil {
+		panic(err)
+	}
+
+	return obj
+}
+
+// ControlPlaneTemplateBuilder holds the variables and objects needed to build a generic template for cluster.spec.controlPlaneRef.
+type ControlPlaneTemplateBuilder struct {
+	namespace                     string
+	name                          string
+	infrastructureMachineTemplate *unstructured.Unstructured
+}
+
+// NewControlPlaneTemplateBuilder creates a NewControlPlaneTemplate builder with the given name and namespace.
+func NewControlPlaneTemplateBuilder(namespace, name string) *ControlPlaneTemplateBuilder {
+	return &ControlPlaneTemplateBuilder{
+		namespace: namespace,
+		name:      name,
+	}
+}
+
+// WithInfrastructureMachineTemplate adds the given Unstructured object to the ControlPlaneTemplateBuilder as its InfrastructureMachineTemplate.
+func (c *ControlPlaneTemplateBuilder) WithInfrastructureMachineTemplate(t *unstructured.Unstructured) *ControlPlaneTemplateBuilder {
+	c.infrastructureMachineTemplate = t
+	return c
+}
+
+// Build creates an Unstructured object from the variables passed to the ControlPlaneTemplateBuilder.
+func (c *ControlPlaneTemplateBuilder) Build() *unstructured.Unstructured {
+	obj := &unstructured.Unstructured{}
+	obj.SetAPIVersion(ControlPlaneGroupVersion.String())
+	obj.SetKind(GenericControlPlaneTemplateKind)
+	obj.SetNamespace(c.namespace)
+	obj.SetName(c.name)
+
+	if err := unstructured.SetNestedField(obj.UnstructuredContent(), true, "spec", "template", "spec", "fakeSetting"); err != nil {
+		panic(err)
+	}
+
+	if c.infrastructureMachineTemplate != nil {
+		if err := setNestedRef(obj, c.infrastructureMachineTemplate, "spec", "template", "spec", "machineTemplate", "infrastructureRef"); err != nil {
```

Comment:
I know we cannot use it right now, but it might make sense to add a comment that this must match the implementation of the internal/contract package?

Reply:
Will add a TODO - these should definitely be updated with the new contract types.